docs/domain.md

```diff
@@ -0,0 +1,107 @@
+files to be completely self-contained pieces of searchable data. It
+is a Xapian single-file "Glass" database, containing a position table, a
+postlist table, a
+record table, etc.
+
+As an example of a metadata file, here's one:
+
+    {
+      "@type": "ekn://_vocab/ImageObject",
+      "@id": "ekn:///7ec07025139a5c22",
+      "@context": "ekn://_context/ImageObject",
+      "copyrightHolder": "Lokilech",
+      "caption": "The Barbarossa chandelier under the dome of the Octagon",
+      "encodingFormat": "image/jpeg",
+      "title": "File:Barbarossaleuchter.jpg",
+      "tags": [
+        "EknMediaObject"
+      ],
+      "language": "en",
+      "license": "GNU Free Documentation License",
+      "sourceURI": "http://upload.wikimedia.org/wikipedia/commons/thumb/2/2a/Barbarossaleuchter.jpg/799px-Barbarossaleuchter.jpg",
+    }
+
+Fields like "@type", "@context" and "@id" are part of [JSON-LD](http://json-ld.org/),
+a specification for hypermedia-style linked data.
+
+## Subscriptions ##
+
+The main content is
+delivered through a mechanism known as "subscriptions". The exact nature and workings of
+subscriptions is out of this document, but it allows content to be updated independently
+from the application.
+
+Each subscription contains a number of shard files, and a
+manifest to list each of the shards as well as various metadata about the
+subscription.
+
+The plumbing that contains the "tie" between an application and its set of subscriptions
+is in a file called `subscriptions.json`, placed in the same directory as
+`EKN_VERSION`. For instance, for the Prensa Libre app, our first subscriptions-based
+application, the `subscriptions.json` file is located in
+`/share/ekn/data/com.endlessm.prensa_libre.es_GT/subscriptions.json`.
+
+The file simply contains a list of subscriptions, like:
+
+```
+{
+    "subscriptions": [
+        {"id": "10521bb3a18b573f088f84e59c9bbb6c2e2a1a67"}
+    ]
+}
+```
+
+While the `subscriptions.json` file gives the appearance that an application can support
+more than one subscription, currently, eos-knowledge-lib only supports one subscription.
+
+The shards are downloaded and stored in the user's home directory, in the directory
+`~/.local/share/com.endlessm.subscriptions/`. Each subscription ID is given its own
+directory for the manifest and shards.
+
+While subscriptions can be downloaded from Endless's servers at any given time, each
+bundle can also come preloaded with a set of subscriptions, packed into the bundle at
+e.g. `/share/ekn/data/com.endlessm.prensa_libre.es_GT/com.endlessm.subscriptions`.
+On first start of
+the application, if missing, the manifest and shards will be symlinked into the user's
+home directory.
+
+At runtime, the user's computer checks for updates to each subscription and downloads new
+shards in the background.
+
+To help inspect applications and subscriptions, eos-knowledge-lib ships
+with a
+tool called `eminem`. Please run `eminem` for more details on usage.
```

Comment:
Yeah, I was trying to leave unchanged lines unchanged in the diff (and rendered in Markdown it doesn't matter), but I changed so much it's not recognized as a rename anymore anyway, so I may as well fix up the wrapping